packages/core/src/endowmentsToolkit.js

```diff
@@ -126,14 +141,49 @@ function endowmentsToolkit({
     )
     // sort by length to optimize further steps
     whitelistedReads.sort((a, b) => a.length - b.length)
-    return makeMinimalViewOfRef(
+    const endowments = makeMinimalViewOfRef(
       sourceRef,
       whitelistedReads,
       unwrapTo,
       unwrapFrom,
       explicitlyBanned,
       allowedWriteFields
     )
+
+    if (allowRedefine.length > 0 && unwrapFrom) {
+      const expectedTarget = unwrapFrom || endowments
+      const originalDefineProperty = unwrapFrom.Object.defineProperty
+      /**
+       * @param {any} target
```

Comment:
```suggestion
       * @param {Record<PropertyKey, unknown>} target
```

or, if TS complains about L168

```suggestion
       * @param {Record<PropertyKey, any>} target
```